test/framework/cluster_proxy.go

```diff
@@ -103,6 +107,21 @@ type ClusterProxy interface {
 	Dispose(context.Context)
 }
 
+// CreateOrUpdateConfig contains options for use with CreateOrUpdate.
+type CreateOrUpdateConfig struct {
+	LabelSelector string
```

Comment:
I don't want to overengineer this, but can't we directly use a `labels.Selector` here? This way we also get rid of the parsing below.

```suggestion
	labelSelector labels.Selector
```

Reply:
yep no reason not to 👍